tests/components/onewire/test_config_flow.py

```diff
@@ -190,6 +198,58 @@ async def test_reconfigure_duplicate(
     assert other_config_entry.data == {CONF_HOST: "2.3.4.5", CONF_PORT: 2345}
 
 
+async def test_hassio_flow(hass: HomeAssistant) -> None:
+    """Test HassIO discovery flow."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN,
+        context={"source": SOURCE_HASSIO},
+        data=_HASSIO_DISCOVERY,
+    )
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "discovery_confirm"
+    assert not result["errors"]
+
+    # Cannot connect to server => retry
+    with patch(
+        "homeassistant.components.onewire.onewirehub.protocol.proxy",
+        side_effect=protocol.ConnError,
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={},
+        )
+
+    assert result["type"] is FlowResultType.FORM
+    assert result["step_id"] == "discovery_confirm"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+    # Connect OK
+    with patch(
+        "homeassistant.components.onewire.onewirehub.protocol.proxy",
+    ):
+        result = await hass.config_entries.flow.async_configure(
+            result["flow_id"],
+            user_input={},
+        )
+
+    assert result["type"] is FlowResultType.CREATE_ENTRY
+    new_entry = result["result"]
+    assert new_entry.title == "owserver (1-wire)"
+    assert new_entry.data == {CONF_HOST: "1302b8e0-owserver", CONF_PORT: 4304}
```

Comment:
Let's also assert the unique id, because I believe that is also set

Reply:
There is no unique_id in onewire

Reply:
but I believe the discovery without unique id does set the domain as unique id

Reply:
https://github.com/home-assistant/core/blob/b1558c7363a74a6ea3d2bd59ff9e6ee5576c851c/homeassistant/config_entries.py#L2784-L2790